### DISCUSSION TITLE
是否有办法从xmake命令向xmake.lua脚本传递参数呢？

### DISCUSSION THREAD

**Discussion by fallending** [xmake-io/xmake#2266] [Q&A] [answered]
### 有关add_configfiles的问题

xmake 提供配置文件生成函数：add_configfiles，但是目前不知道如何从命令中控制一些变量，最终产生到 configfile中，求教～

**Comment by waruqi** [comment, marked as answer]
用 option

**Reply by fallending** [reply]
* cmdline
```
# Manually enable this option
$ xmake f --test=y
$ xmake
```
* xmake.lua
```
option("test")
    set_default(false)
    set_showmenu(true)
    add_defines("TEST")
```

[option doc](https://xmake.io/#/manual/configuration_option?id=defining-options)
